File: backend/src/controllers/gerenciadorAcessoController.js
const GerenciadorAcesso = require('../models/GerenciadorAcesso');

// Buscar acesso por CNPJ
exports.getAcesso = async (req, res) => {
  try {
    const { cnpj } = req.params;
    const doc = await GerenciadorAcesso.findOne({ cnpj });
    if (!doc) return res.json({ link: '', login: '', senha: '' });
    res.json(doc);
  } catch (err) {
    res.status(500).json({ error: 'Erro ao buscar acesso', details: err.message });
  }
};

// Salvar acesso (cria ou atualiza)
exports.saveAcesso = async (req, res) => {
  try {
    const { cnpj } = req.params;
    const { link, login, senha, atualizadoPor } = req.body;
    const doc = await GerenciadorAcesso.findOneAndUpdate(
      { cnpj },
      { $set: { link, login, senha, atualizadoPor, atualizadoEm: new Date() } },
      { upsert: true, new: true }
    );
    res.json(doc);
  } catch (err) {
    res.status(500).json({ error: 'Erro ao salvar acesso', details: err.message });
  }
};

File: python_backend/controllers/gerenciador_acesso_controller.py
from db import db
from fastapi import HTTPException
from pymongo import ReturnDocument

async def get_acesso(cnpj: str):
    coll = db['gerenciador_acesso']
    doc = await coll.find_one({'cnpj': cnpj})
    if not doc:
        return {'link': '', 'login': '', 'senha': ''}
    doc['_id'] = str(doc.get('_id'))
    return doc

async def save_acesso(cnpj: str, payload: dict):
    link = payload.get('link', '')
    login = payload.get('login', '')
    senha = payload.get('senha', '')
    atualizadoPor = payload.get('atualizadoPor')
    coll = db['gerenciador_acesso']
    res = await coll.find_one_and_update({'cnpj': cnpj}, {'$set': {'link': link, 'login': login, 'senha': senha, 'atualizadoPor': atualizadoPor, 'atualizadoEm': __import__('datetime').datetime.utcnow()}}, upsert=True, return_document=ReturnDocument.AFTER)
    if res:
        res['_id'] = str(res.get('_id'))
    return res
